### PR TITLE
Gun adapter improvements

### PR DESCRIPTION
**Pull request by alex-strizhakov**
- checking original url in adapter code
- flush stream on max_body error
- removed duplicate test
- removed `original` option in tests
- set `tls` transport for all `https` requests

### CODE REVIEW

**Comment by rinpatch** [comment]
Also mixed feelings about the way the adapter handles redirects. If the user passes an existing connection to the adapter, it's probably because they are using a connection pool, so opening a new off-pool connection doesn't feel like the right thing to do. Maybe allow passing a callback that gets a connection to the specified host and only open a connection if it's not present

File: lib/tesla/adapter/gun.ex [hunk 1]
```diff
@@ -162,8 +161,32 @@ if Code.ensure_loaded?(:gun) do
       end
     end
 
-    defp check_original(%URI{host: host, port: port}, %{original: original} = opts) do
-      Map.put(opts, :original_matches, original == "#{domain_or_fallback(host)}:#{port}")
+    defp check_original(%URI{host: host, port: port, scheme: scheme}, %{conn: conn} = opts) do
+      matches? =
+        try do
+          # sometimes :gun.info can raise MatchError
```
[the remark this answers is not in the document]
Comment (alex-strizhakov) [reply lib/tesla/adapter/gun.ex]: Saw it in the logs few times. Maybe it can be removed now.

**Comment by teamon** [comment]
Hey @alex-strizhakov, how's going with this PR? :)

**Comment by alex-strizhakov** [comment]
hey @teamon!
PR is ready and tested, if you don't have any comments. Option with callback for redirects i'll make later in separate PR.

**Pull request merged**